libminifi/include/core/FlowFile.h

```diff
@@ -310,6 +311,10 @@ struct SpecialFlowAttribute {
   static const std::string ALTERNATE_IDENTIFIER;
   // Flow identifier
   static const std::string FLOW_ID;
+
+  static std::unordered_set<std::string> getSpecialFlowAttributes() {
+    return {PATH, ABSOLUTE_PATH, FILENAME, UUID, priority, MIME_TYPE, DISCARD_REASON, ALTERNATE_IDENTIFIER, FLOW_ID};
+  }
```

Comment:
I'm not sure it's a good idea as there are currently 69 occurrences of the `SpecialFlowAttribute` in the codebase and it would take a lot of changes. My other problem with it is that the main use case of the `SpecialFlowAttribute`s is using them as strings while getting or setting attributes so the enums would always have to be converted to strings which would make their usage harder.

Reply:
OK, makes sense.  I thought it was already an enum, just not a smart one, but you're right, it's not.